static/js/narrow_banner.js

```diff
@@ -36,32 +29,50 @@ function show_search_query() {
             stream_topic_string = stream_topic_string + " topic: " + topic;
         }
 
-        search_string_display.append(" ");
-        search_string_display.append($("<span>").text(stream_topic_string));
+        search_string_display += " ";
+        search_string_display += `<span id="empty_search_topic">${_.escape(
+            stream_topic_string,
+        )}</span>`;
```

Comment:
I suspect that this templating logic would be cleaner with a Handlebars template, in part because that'd take care of the escaping more naturally for us, as `.text` did previously.  IS there a reason you migrated to needing to explicitly escape things like this?

Reply:
So I thought of making the template renderer utility function as simple and generic as possible, without having custom logic to append search query strings (like having an extra optional parameter like search_queries). What do you suggest?

Reply:
I think it's very reasonable for it to have a couple conditionals for things like this where we'd otherwise be repeating a lot of code.